hs/src/Reach/Backend/JS.hs

```diff
@@ -456,20 +456,20 @@ jsExpr = \case
             [ ("amt", zero')
             , ("tok", tok')
             ]
-  DLE_TokenAccepted _ acc tok -> do
+  DLE_TokenAccepted _ addr tok -> do
     (ctxt_mode <$> ask) >>= \case
-      JM_Backend -> do
-        -- TODO - This is wrong, but I'm still not sure what to do here.
-        return $ "false"
+      JM_Backend -> return "true"
       JM_View -> impossible "view canReceive"
       JM_Simulate -> do
-        acc' <- jsArg acc
+        addr' <- jsArg addr
         tok' <- jsArg tok
+        let simTxn = jsSimTxn "tokenAccepted" $ [("tok", tok'), ("addr", addr')]
+        -- TODO - this doesn't work because addr' is a JS string, so it doesn't have a tokenAccepted method.  How do I transform it into an address here?  Or am I going about this the wrong way?
+        let accepted = "await" <+> jsApply (addr' <> ".tokenAccepted") [tok']
         return $
-          jsSimTxn "tokenAccepted" $
-            [ ("tok", tok')
-            , ("acc", acc')
-            ]
+          "await" <+> jsApply
+          (parens $ "async () => " <>
+           (jsBraces $ vsep [simTxn, "return" <+> accepted])) []
```

Comment:
This is not necessary. You are already in an async context. If you're just doing this because you want to have the effect, my advice is to (a) use the comma operator or (b) have the `simTokenAccepted` function take the effect of `jsSimTxn` as an argument (which is ignored)